src/cascadia/TerminalControl/TermControlAutomationPeer.cpp

```diff
@@ -248,7 +253,7 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         dispatcher.RunAsync(Windows::UI::Core::CoreDispatcherPriority::Normal, [weakThis{ get_weak() }, sanitizedCopy{ hstring{ sanitized } }]() {
             if (auto strongThis{ weakThis.get() })
             {
-                if (auto control{ strongThis->_termControl.get() })
+                if (auto control{ strongThis->_termControl.get() }; control && !strongThis->_closing)

```

Comment:
@lhecker do we need some sort of atomic for `_closing` here? We're guaranteed multiple threads will contend for it.

Reply:
The `_closing` check is ineffective even if it was atomic, because TermControl might close on the main thread while this background thread is running the code inside the if condition. In other words, the if condition would be about as effective if the `_closing` condition wouldn't exist at all.

...but why does the `_closing` check exist at all? The PR message implies that all we need to do is to keep TermControl and TCAP alive. But the old code already did that, so is the PR message incomplete?

Solving both problems may be done by using a semaphore (= unblock when the count is 0). Unfortunately `std::semaphore` doesn't work for this purpose (it blocks when the count is 0), but an alternative may be built with `std::condition_variable` or similar.

Reply:
The problem is, the UIA machinery in XAML doesn't care whether our object is still alive. We need an external signal that we should stop attempting to produce notifications... we don't want to wait for them to finish